src/broker/broker_config.c

```diff
@@ -652,19 +655,21 @@ broker_config_read_internal (const char *conf_file, T_BROKER_INFO * br_info, int
 	}
 
       br_info[num_brs].appl_server_min_num =
-	ini_getuint (ini, sec_name, "MIN_NUM_APPL_SERVER", DEFAULT_AS_MIN_NUM, &lineno);
-      br_info[num_brs].appl_server_num = br_info[num_brs].appl_server_min_num;
-      if (br_info[num_brs].appl_server_min_num > APPL_SERVER_NUM_LIMIT)
+	ini_getint (ini, sec_name, "MIN_NUM_APPL_SERVER", DEFAULT_AS_MIN_NUM, &lineno);
+      if (br_info[num_brs].appl_server_min_num < 1)
 	{
-	  errcode = PARAM_BAD_VALUE;
+	  errcode = PARAM_BAD_RANGE;
 	  goto conf_error;
 	}
 
+      br_info[num_brs].appl_server_num = br_info[num_brs].appl_server_min_num;
+
       br_info[num_brs].appl_server_max_num =
-	ini_getuint (ini, sec_name, "MAX_NUM_APPL_SERVER", DEFAULT_AS_MAX_NUM, &lineno);
-      if (br_info[num_brs].appl_server_max_num > APPL_SERVER_NUM_LIMIT)
+	ini_getint (ini, sec_name, "MAX_NUM_APPL_SERVER", DEFAULT_AS_MAX_NUM, &lineno);
+      if (br_info[num_brs].appl_server_max_num > APPL_SERVER_NUM_LIMIT ||
+	  br_info[num_brs].appl_server_max_num < br_info[num_brs].appl_server_min_num)
```

Comment:
How about returning PARAM_BAD_RANGE instead of PARAM_BAD_MAX_APPL_SERVER if app_server_man_num is greater than APPL_SERVER_NUM_LIMIT ?
When user meet that error, I think that user can mis-understand.

Reply:
How about handling the error by dividing the condition of the if statement into two?
If appl_server_max_num > APPL_SERVER_NUM_LIMIT, return PARAM_BAD_RANGE.
If appl_server_max_num < appl_server_min_num, return PARAM_BAD_MAX_APPL_SERVER.

Reply:
It's a little bit complicated. My suggestion is:

1. if appl_server_max_num > APPL_SERVER_NUM_LIMIT or appl_server_max_num < 1
   return PARAM_BAD_RANGE
2. appl_server_max_num < appl_server_min_num
   return PARAM_BAD_MAX_APPL_SERVER
   with the message:

   "Value is out of range (MAX_NUM_APPL_SERVER must be greater than MIN_NUM_APPL_SERVER)."

Please comment!

Reply:
I think this suggestion is good.